### ISSUE TITLE
Magnetic Tilt and Multi-Scale Tilt Depth Estimation

### ISSUE THREAD

**Issue by mtb-za** [fatiando/fatiando#203]
Hi all,

I am looking at implementing a method (Multi-Scale Tilt Depth Estimation) developed by Reece van Buren in 2009, which he used as the basis for an MSc thesis with Dr Gordon Cooper at the University of the Witwatersrand in 2013.

As a prerequisite, this requires implementing the tilt ( `T(f) = tan^(-1)[(df/dz)/sqrt((df/dx)^2)]` ) which Miller and Singh defined in 1994.

Are there any objections to me doing so? Is tilt already done, but I am using old nomenclature? I can not find any references to it in the codebase, but may have missed something.

## References
Van Buren, Reece. 2013. “Multi-Scale Tilt Depth Estimation.” MSc Thesis,
Johannesberg: University of the Witwatersrand.
URI: [wits.ac.za](http://mobile.wiredspace.wits.ac.za/bitstream/handle/10539/14011/MSc_Dissertation_RvanBuren_2013.pdf).

Van Buren, Reece. 2009. “Multi-Scale Depth Estimation.”
In 11th SAGA Biennial Technical Meeting and Exhibition, 236–40. Swaziland.

**Comment by leouieda** [comment]
@mtb-za neither of these have been implemented already. You're more than welcome to have a go!

The tilt angle can be a part of `gravmag.transform` and should be too much work because we already have the derivatives in there (and once #196 is merged they will be even better).

The depth estimation can go in there as well, I guess. Probably no point in creating a new module for just one function. If we gather more depth estimation functions later we can move them to a new module.

**Comment by mtb-za** [comment]
Cool. Will give that a stab.

**Comment by leouieda** [comment]
You can use the `derivx`, `derivz` functions in `gravmag.transform`.  They'll work for any gridded data.

Also, I invited you to join the Github org for Fatiando. This way, you'll show up on the group page and the group  will appear on your profile. Plus, I can assign you to issues and PRs.

**Comment by mtb-za** [comment]
Just to confirm then, `derivx` and `derivz` will get the `df/dx` and `df/dz`?

That would mean that tilt would be implemented by something like:

    def tilt(x, y, data, shape):
       horiz_deriv = derivx(x, y, data, shape)/derivy(x, y, data, shape)
       tilt = atan( ( derivz(x, y, data, shape) ) / sqrt( exp(horiz_deriv,2) ) )
       return tilt

Although, I will need to check that the `atan` does something sensible....
